pkg/deploy/devfile_registry.go

```diff
@@ -27,49 +27,90 @@ type DevFileRegistryConfigMap struct {
 }
 
 const (
-	DevfileRegistry = "devfile-registry"
+	DevfileRegistry              = "devfile-registry"
+	devfileRegistryGatewayConfig = "devfile-registry-gtw"
 )
 
 /**
  * Create devfile registry resources unless an external registry is used.
  */
-func SyncDevfileRegistryToCluster(deployContext *DeployContext) (bool, error) {
+func SyncDevfileRegistryToCluster(deployContext *DeployContext, cheHost string) (bool, error) {
 	devfileRegistryURL := deployContext.CheCluster.Spec.Server.DevfileRegistryUrl
 	if !deployContext.CheCluster.Spec.Server.ExternalDevfileRegistry {
 		var host string
+		exposureStrategy := util.GetServerExposureStrategy(deployContext.CheCluster, DefaultServerExposureStrategy)
+		singleHostExposureType := GetSingleHostExposureType(deployContext.CheCluster)
+		useGateway := exposureStrategy == "single-host" && (util.IsOpenShift || singleHostExposureType == "gateway")
```

Comment:
do we really need this? on the places we want to ask whether to use gateway, we should already know that we're on single-host right? Also it should not be possible to make it here on openshift with `exposureStrategy` set to `single-host` and `singleHostExposureType` set to anything different than `gateway`. Such configuration should fail earlier.